### PR TITLE
Refactor timeline architecture.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
No added functionality. Refactored the structure of the timeline code to make it more maintainable and robust. It now follows this pattern:

DevTools Timeline Architecture

Key parts:
**View**: The view has no awareness of the timeline data directly. It has access to the timeline data (frames, selected event, selected frame, cpu profile, etc.) through the `TimelineController`. It receives updates from the `TimelineController` via streams.

**Model**: `TimelineData` stores the current data for the DevTools timeline. It contains `CpuProfileData` and is updated by `TimelineController`

**Controller**:  The controller manages the timeline data model and communicates with the view to give and receive data updates. It manages data processing via protocols `TimelineProtocol` (protocol for processing trace events and composing them into `TimelineEvents` and `TimelineFrames`) and `CpuProfileProtocol` (protocol for processing `CpuProfileData` and composing it into a structured tree of `CpuStackFrame`'s). The controller also communicates with `TimelineService`, which manages interactions between the Timeline and the VmService. `TimelineController` has no dependency on `dart:html`, making it more easily testable.
 
This CL also improves test coverage.

### CODE REVIEW

File: packages/devtools/lib/src/framework/framework.dart [hunk 1]
```diff
@@ -6,12 +6,13 @@ import 'dart:async';
 import 'dart:convert';
 import 'dart:html' hide Screen;
 
+import 'package:devtools/src/timeline/timeline_model.dart';
```
Comment (devoncarew) [inline packages/devtools/lib/src/framework/framework.dart]: drive by comment: I'd use a relative path import there

**Comment by devoncarew** [comment]
I'll defer to others for the review, but the overall arch. change looks great.

File: packages/devtools/lib/src/framework/framework.dart (hunk 1 above)
Comment (terrylucas) [reply packages/devtools/lib/src/framework/framework.dart]: Sorted?

File: packages/devtools/lib/src/timeline/cpu_profile_model.dart [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+
+import '../utils.dart';
+
+/// Data model for DevTools CPU profile.
+class CpuProfileData {
+  CpuProfileData._({
+    this.stackFramesJson,
```
Comment (terrylucas) [inline packages/devtools/lib/src/timeline/cpu_profile_model.dart]: Should all of these params be @required?

File: packages/devtools/lib/src/timeline/cpu_profile_model.dart [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  });
+
+  final String id;
+  final String name;
+  final String category;
+  final String url;
+
+  CpuStackFrame parent;
+
+  List<CpuStackFrame> children = [];
```
Comment (terrylucas) [inline packages/devtools/lib/src/timeline/cpu_profile_model.dart]: Final?

File: packages/devtools/lib/src/ui/theme.dart [hunk 4]
```diff
@@ -54,6 +54,10 @@ const defaultButtonIconColor = ThemedColor(
 class ThemedColor implements Color {
   const ThemedColor(this._light, this._dark);
 
+  static ThemedColor fromSingleColor(Color color) {
```
Comment (terrylucas) [inline packages/devtools/lib/src/ui/theme.dart]: nit: =>

**Pull request merged**